src/ResourceManagement/Network/Microsoft.Azure.Management.Network.Fluent/ApplicationGatewayListenerImpl.cs

```
@@ -0,0 +1,290 @@
+            };
+
+            Inner.FrontendIPConfiguration = frontendRef;
+            return this;
+        }
+
+        ///GENMHASH:85408D425EF4341A6D39C75F68ED8A2B:60CB6094D3C3120ABC72B2E26313EF5A
+        public ApplicationGatewayListenerImpl WithServerNameIndication()
+        {
+            Inner.RequireServerNameIndication = true;
+            return this;
+        }
+
+        ///GENMHASH:604F12B361C77B3E3AD5768A73BA6DCF:ABADA5A8E77FDA08CD893ADDC4895F32
+        public ApplicationGatewayListenerImpl WithHttp()
+        {
+            Inner.Protocol = ApplicationGatewayProtocol.Http.ToString();
+            return this;
+        }
+
+        ///GENMHASH:AFBFDB5617AA4227641C045CF9D86F66:51AAC8329531E69BAAFAC03299414449
+        public ApplicationGatewayListenerImpl WithSslCertificateFromPfxFile(FileInfo pfxFile)
+        {
+            return WithSslCertificateFromPfxFile(pfxFile, null);
+        }
+
+        ///GENMHASH:8ADE3A1B894E5D01B188A0822FC89126:C9DB02E58BCE0380CFA35181A84B25CD
+        private ApplicationGatewayListenerImpl WithSslCertificateFromPfxFile(FileInfo pfxFile, string name)
+        {
+            if (name == null)
+            {
+                name = ResourceNamer.RandomResourceName("cert", 10);
+            }
+            sslCert = Parent.DefineSslCertificate(name)
+                .WithPfxFromFile(pfxFile);
+            return this;
+        }
+
+        ///GENMHASH:382D2BF4EBC04F5E7DF95B5EF5A97146:66502B9267EB235997FE645B0C0A6527
+        public ApplicationGatewayListenerImpl WithSslCertificatePassword(string password)
+        {
+            if (sslCert != null)
+            {
+                sslCert.WithPfxPassword(password).Attach();
+                WithSslCertificate(sslCert.Name());
+                sslCert = null;
+                return this;
+            }
+            else
+            {
+                return null; // Fail fast as this should never happen if the internal logic is correct
```

Agree with Fail fast. Do not agree with NullRef exception. better throw ArgumentException with descriptive message.